datasets/electricity/electricity.py

```diff
@@ -0,0 +1,217 @@
+                # "feat_dynamic_real":  datasets.Sequence(datasets.Sequence(datasets.Value("uint64"))),
+                # "feat_dynamic_cat": datasets.Sequence(datasets.Sequence(datasets.Value("uint64"))),
+                "item_id": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features, uncomment supervised_keys line below and
+            # specify them. They'll be used if as_supervised=True in builder.as_dataset.
+            # supervised_keys=("sentence", "label"),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLS
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        data_dir = dl_manager.download_and_extract(_URL)
+
+        # define the prediction problem # TODO save these in metadata
+        freq = "1H"
+        prediction_length = 24
+        rolling_evaluations = 7
+
+        train_ts = []
+        val_ts = []
+        test_ts = []
+
+        df = pd.read_csv(
+                Path(data_dir) / "LD2011_2014.txt",
+                sep=";",
+                index_col=0,
+                parse_dates=True,
+                decimal=",",
+            )
+        df.sort_index(inplace=True)
+        df = df.resample(freq).sum()
+            
+        if self.config.name == "uci":
+            val_end_date = df.index.max() - pd.Timedelta(prediction_length * rolling_evaluations, "H")
+            train_end_date = val_end_date - pd.Timedelta(prediction_length, "H")
+        else:
+            # concate the time series to be from 2012 till 2014
+            df = df[(df.index.year >= 2012) & (df.index.year <= 2014)]
+
+            # drop time series which are zero at the start
+            df = df.T[df.iloc[0] > 0].T
+
+            # validation ends at 8/10-th of the time series
+            val_end_date = df.index[int(len(df) * (8 / 10)) - 1]
+            # training ends at 6/10-th of the time series
+            train_end_date = df.index[int(len(df) * (6 / 10)) - 1]
```

Comment:
How did you come up with these proportions for the splits?  Ideally, all the data should be in the train split, unless there are predefined splits.

Another option is to add these values to the config to be customizable:
```
val_proportion = 8 / 10
train_proportion = 6 / 10
```